src/python/tests/fasta/test_chunk.py

```diff
@@ -0,0 +1,441 @@
+                    None,
+                    n_sequece_len,
+                    chunk_sfx,
+                    append_offset_to_chunk_name,
+                    open_individual=_individual_opener,
+                )
+            assert output_fasta.getvalue() == expected_chunked_fasta_text
+            assert agp_list == expected_agp_list
+            assert len(parts) == 0
+
+    with StringIO(input_fasta_text) as input_fasta:
+        with nullcontext() as no_output_fasta:
+            parts.clear()
+            with expected_raised:
+                agp_list = FastaChunking.chunk_fasta_stream(
+                    input_fasta,
+                    chunk_size,
+                    chunk_size_tolerated,
+                    no_output_fasta,
+                    "/path/prefix",
+                    n_sequece_len,
+                    chunk_sfx,
+                    append_offset_to_chunk_name,
+                    open_individual=_individual_opener,
+                )
+            assert "".join(map(lambda p: p[1], parts)) == expected_chunked_fasta_text
+            assert agp_list == expected_agp_list
+            assert len(parts) == expected_individual_files_count
+
+
+@pytest.mark.parametrize(
+    "individual_file_prefix, agp_output_file_name, expected_missing_joined",
+    [
+        ("chunk", "test.agp", pytest.raises(FileNotFoundError, match=r"output.fa")),
+        ("chunk", None, pytest.raises(FileNotFoundError, match=r"output.fa")),
+        ("", "test.agp", does_not_raise()),
+        (None, "test.agp", does_not_raise()),
+        ("", "", does_not_raise()),
+        (None, None, does_not_raise()),
+    ],
+)
+def test_chunk_fasta(
+    monkeypatch: Any,
+    tmp_path: Path,
+    individual_file_prefix: Optional[str],
+    agp_output_file_name: Optional[str],
+    expected_missing_joined: ContextManager,
+) -> None:
+    """Tests the `chunk.chunk_fasta` function.
+
+    Args:
+        tmp_path: Where temporary files will be created.
+        individual_file_prefix: A file path prefix including dirs and filenames part to use as a
+                firts part of the chunk file name or None.
+        agp_output_file_name: Output AGP file name or None.
+        expected_missing_joined: A context manager with expected exception (`pytest.raises` or nullcontext).
+    """
+
+    # a helper mock function
+    def _chunk_fasta_stream_mock(
+        input_fasta: TextIOWrapper,
+        chunk_size: int,
+        chunk_size_tolerated: int,
+        output_fasta: Optional[TextIOWrapper] | nullcontext[Any],
+        individual_file_prefix: Optional[str],
+        n_sequece_len: int,
+        chunk_sfx: str,
+        append_offset_to_chunk_name: Optional[bool],
+        open_individual: Callable[[str], TextIOWrapper] = FastaChunking._individual_file_opener,
+    ) -> list[str]:
+        """Mock the `chunk.chunk_fasta_stream` function.
+
+        Args:
+            *args: Positional argumnets.
+            **kwargs: Keyword arguments.
```

Comment:
Either add the docstring for all the arguments or skip it since this is an internal function.